src/sparseml/exporters/onnx_to_deepsparse.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import onnx
+
+from sparseml.exporters import transforms as sparseml_transforms
+from sparseml.exporters.base_exporter import BaseExporter
+
+
+class ONNXToDeepsparse(BaseExporter):
+    """
+    Optimizes an `onnx.ModelProto` for the deepsparse engine by applying a
+    series of transformations to a onnx graph with quantize operations.
+
+    Usage:
+    ```python
+    model: onnx.ModelProto = ... # could be from TorchToOnnx() or somewhere else
+    exporter = ONNXToDeepsparse()
+    exporter.export(model, "model.onnx")
+    ```
+
+    You can also just optimize the model directly without saving to disk:
+    ```python
+    model: onnx.ModelProto = ...
+    exporter = ONNXToDeepsparse()
+    optimized_model = exporter.apply(model)
+    ```
+
+    :param use_qlinearconv: Set True to use legacy QLinearConv format instead
+        of ConvInteger. QLinearConv requires output activations be quantized
+        in the quantization recipe. (This was the default behavior prior to
+        sparseml 0.12). Default is False
+    :param skip_input_quantize: if True, the export flow will attempt to delete
+        the first Quantize Linear Nodes(s) immediately after model input and set
+        the model input type to UINT8. Default is False
+    """
+
+    def __init__(
+        self,
+        use_qlinear_conv: bool = False,
+        skip_input_quantize: bool = False,
+    ) -> None:
+        conv_transform = (
+            sparseml_transforms.ConvToQLinearConv()
+            if use_qlinear_conv
+            else sparseml_transforms.ConvertQuantizableConvInteger()
+        )
+
+        transforms = [
```

Comment:
let's think about splitting this up into the first half and the actual Q/DQ conversions - first half is mostly around cleaning up the torch export graph